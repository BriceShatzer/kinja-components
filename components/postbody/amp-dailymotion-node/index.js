// @flow

import AmpDailyMotionNode from './amp-dailymotion-node';

export default AmpDailyMotionNode;
