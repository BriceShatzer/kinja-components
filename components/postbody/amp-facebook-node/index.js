// @flow

import AmpFacebookNode from './amp-facebook-node';
import AmpFacebookPostNode from './amp-facebookPost-node';

export {
	AmpFacebookNode,
	AmpFacebookPostNode
};