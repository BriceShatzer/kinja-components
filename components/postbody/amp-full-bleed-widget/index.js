// @flow

import AmpFullBleedWidget from './amp-full-bleed-widget';

export default AmpFullBleedWidget;
