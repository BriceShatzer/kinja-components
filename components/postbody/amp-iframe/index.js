// @flow

import AmpIframe from './amp-iframe';

export default AmpIframe;
