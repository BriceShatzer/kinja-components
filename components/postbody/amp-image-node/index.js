// @flow

import AmpImageNode from './amp-image-node';

export default AmpImageNode;
