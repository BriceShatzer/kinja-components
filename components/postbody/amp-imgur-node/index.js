// @flow

import AmpImgurNode from './amp-imgur-node';

export default AmpImgurNode;
