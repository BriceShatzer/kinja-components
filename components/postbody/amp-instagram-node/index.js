// @flow

import AmpInstagramNode from './amp-instagram-node';

export default AmpInstagramNode;
