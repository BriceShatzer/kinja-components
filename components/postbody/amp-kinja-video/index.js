import AmpKinjaVideo from './amp-kinja-video';

export default AmpKinjaVideo;
