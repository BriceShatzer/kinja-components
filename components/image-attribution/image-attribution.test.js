/* eslint-disable flowtype/no-types-missing-file-annotation */

import * as React from 'react';

import { shallow, render } from 'enzyme';

import { TextNode, LinkNode } from 'postbody/InlineNode';
import ImageAttributionComponent, { Licensors } from './image-attribution';

import type { InlineNode } from 'postbody/InlineNode';
import type { AttributionLabel, ImageAttribution } from 'postbody/Image';

function attribute(label?: AttributionLabel, credit?: Array<InlineNode>, source?: Array<InlineNode>): ImageAttribution {
	return {
		label: label ? label : 'Illustration',
		credit: credit ? credit : [new TextNode('credit')],
		source: source ? source : [new TextNode('source')]
	};
}

describe('<ImageAttributionComponent />', () => {

	test('should not render anything if there is no attribution', () => {
		expect(shallow(<ImageAttributionComponent attributions={[]} />)).toMatchSnapshot();
	});

	test('should render image attributions with credit and source', () => {
		const attribution = [
			attribute(undefined, [new TextNode('credit 1')], [new TextNode('source 1')]),
			attribute('Graphic', [new TextNode('credit 2')], [new TextNode('source 2')]),
			attribute(undefined, [new TextNode('credit 3')], [new TextNode('source 3')])
		];

		expect(shallow(<ImageAttributionComponent attributions={attribution} />)).toMatchSnapshot();
	});

	test('should render image attributions with credit only', () => {
		const attribution = [
			attribute(undefined, [new TextNode('credit 1')], []),
			attribute('Graphic', [new TextNode('credit 2')], []),
			attribute(undefined, [new TextNode('credit 3')], [])
		];

		expect(shallow(<ImageAttributionComponent attributions={attribution} />)).toMatchSnapshot();
	});

	test('should render image attributions with source only', () => {
		const attribution = [
			attribute(undefined, [], [new TextNode('source 1')]),
			attribute('Graphic', [], [new TextNode('source 2')]),
			attribute(undefined, [], [new TextNode('source 3')])
		];

		expect(shallow(<ImageAttributionComponent attributions={attribution} />)).toMatchSnapshot();
	});

	test('should render licensors', () => {
		const attribution = [attribute(), attribute(undefined, [], [new TextNode('source 1')])];

		expect(shallow(<Licensors attributions={attribution} />)).toMatchSnapshot();
	});

	test('should render every licensor once', () => {
		const attribution = [attribute(), attribute(), attribute()];

		expect(shallow(<Licensors attributions={attribution} />)).toMatchSnapshot();
	});

	test('should render for the editor and tokenize', () => {
		const anchor = new LinkNode([new TextNode('I am a link')], 'gawker.com');
		const attribution = [attribute(), attribute(null, [anchor]), attribute(null, null, [anchor])];

		expect(render(<ImageAttributionComponent attributions={attribution} tokenize />)).toMatchSnapshot();
	});

	test('should render and not tokenize by defult', () => {
		const anchor = new LinkNode([new TextNode('I am a link')], 'gawker.com');
		const attribution = [attribute(), attribute(null, [anchor]), attribute(null, null, [anchor])];

		expect(render(<ImageAttributionComponent attributions={attribution} />)).toMatchSnapshot();
	});

});
