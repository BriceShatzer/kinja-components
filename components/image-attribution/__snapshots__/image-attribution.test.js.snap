// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageAttributionComponent /> should not render anything if there is no attribution 1`] = `<Fragment />`;

exports[`<ImageAttributionComponent /> should render and not tokenize by defult 1`] = `
Array [
  "Illustration: credit (source), ",
  .c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  color: #222;
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  box-shadow: inset 0 -2px 0 #0A7BC2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  color: #0A7BC2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<span>
    <a
      class="c0 js_link c1"
      data-ga="[[\\"Embedded Url\\",\\"Internal link\\",\\"\\",{\\"metric25\\":1}]]"
      href=""
    >
      I am a link
    </a>
  </span>,
  " (source), credit (",
  .c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  color: #222;
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  box-shadow: inset 0 -2px 0 #0A7BC2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  color: #0A7BC2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<span>
    <a
      class="c0 js_link c1"
      data-ga="[[\\"Embedded Url\\",\\"Internal link\\",\\"\\",{\\"metric25\\":1}]]"
      href=""
    >
      I am a link
    </a>
  </span>,
  ")",
]
`;

exports[`<ImageAttributionComponent /> should render every licensor once 1`] = `
<WithKinjaMeta
  key="0"
  nodes={
    Array [
      TextNode {
        "styles": Array [],
        "type": "Text",
        "value": "source",
      },
    ]
  }
/>
`;

exports[`<ImageAttributionComponent /> should render for the editor and tokenize 1`] = `
.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  color: #222;
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  box-shadow: inset 0 -2px 0 #0A7BC2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  color: #0A7BC2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<span
  data-label="Illustration"
>
  Illustration: 
  <span
    class="item"
  >
    <span
      class="credit"
    >
      credit
    </span>
    <span
      class="source"
    >
       (source)
    </span>
  </span>
  , 
  <span
    class="item"
  >
    <span
      class="credit"
    >
      <a
        class="c0 js_link c1"
        href=""
      >
        I am a link
      </a>
    </span>
    <span
      class="source"
    >
       (source)
    </span>
  </span>
  , 
  <span
    class="item"
  >
    <span
      class="credit"
    >
      credit
    </span>
    <span
      class="source"
    >
       (
      <a
        class="c0 js_link c1"
        href=""
      >
        I am a link
      </a>
      )
    </span>
  </span>
</span>
`;

exports[`<ImageAttributionComponent /> should render image attributions with credit and source 1`] = `
<Fragment>
  <Token
    data-label="Illustration"
    tokenize={false}
  >
    Illustration
    : 
    <Token
      className="item"
      key="0"
      tokenize={false}
    >
      <AttributionItem
        credit={
          Array [
            TextNode {
              "styles": Array [],
              "type": "Text",
              "value": "credit 1",
            },
          ]
        }
        key="0"
        source={
          Array [
            TextNode {
              "styles": Array [],
              "type": "Text",
              "value": "source 1",
            },
          ]
        }
        tokenize={false}
      />
    </Token>
    , 
    <Token
      className="item"
      key="1"
      tokenize={false}
    >
      <AttributionItem
        credit={
          Array [
            TextNode {
              "styles": Array [],
              "type": "Text",
              "value": "credit 3",
            },
          ]
        }
        key="1"
        source={
          Array [
            TextNode {
              "styles": Array [],
              "type": "Text",
              "value": "source 3",
            },
          ]
        }
        tokenize={false}
      />
    </Token>
  </Token>
  , 
  <Token
    data-label="Graphic"
    tokenize={false}
  >
    Graphic
    : 
    <Token
      className="item"
      key="0"
      tokenize={false}
    >
      <AttributionItem
        credit={
          Array [
            TextNode {
              "styles": Array [],
              "type": "Text",
              "value": "credit 2",
            },
          ]
        }
        key="0"
        source={
          Array [
            TextNode {
              "styles": Array [],
              "type": "Text",
              "value": "source 2",
            },
          ]
        }
        tokenize={false}
      />
    </Token>
  </Token>
</Fragment>
`;

exports[`<ImageAttributionComponent /> should render image attributions with credit only 1`] = `
<Fragment>
  <Token
    data-label="Illustration"
    tokenize={false}
  >
    Illustration
    : 
    <Token
      className="item"
      key="0"
      tokenize={false}
    >
      <AttributionItem
        credit={
          Array [
            TextNode {
              "styles": Array [],
              "type": "Text",
              "value": "credit 1",
            },
          ]
        }
        key="0"
        source={Array []}
        tokenize={false}
      />
    </Token>
    , 
    <Token
      className="item"
      key="1"
      tokenize={false}
    >
      <AttributionItem
        credit={
          Array [
            TextNode {
              "styles": Array [],
              "type": "Text",
              "value": "credit 3",
            },
          ]
        }
        key="1"
        source={Array []}
        tokenize={false}
      />
    </Token>
  </Token>
  , 
  <Token
    data-label="Graphic"
    tokenize={false}
  >
    Graphic
    : 
    <Token
      className="item"
      key="0"
      tokenize={false}
    >
      <AttributionItem
        credit={
          Array [
            TextNode {
              "styles": Array [],
              "type": "Text",
              "value": "credit 2",
            },
          ]
        }
        key="0"
        source={Array []}
        tokenize={false}
      />
    </Token>
  </Token>
</Fragment>
`;

exports[`<ImageAttributionComponent /> should render image attributions with source only 1`] = `
<Fragment>
  <Token
    data-label="Illustration"
    tokenize={false}
  >
    Illustration
    : 
    <Token
      className="item"
      key="0"
      tokenize={false}
    >
      <AttributionItem
        credit={Array []}
        key="0"
        source={
          Array [
            TextNode {
              "styles": Array [],
              "type": "Text",
              "value": "source 1",
            },
          ]
        }
        tokenize={false}
      />
    </Token>
    , 
    <Token
      className="item"
      key="1"
      tokenize={false}
    >
      <AttributionItem
        credit={Array []}
        key="1"
        source={
          Array [
            TextNode {
              "styles": Array [],
              "type": "Text",
              "value": "source 3",
            },
          ]
        }
        tokenize={false}
      />
    </Token>
  </Token>
  , 
  <Token
    data-label="Graphic"
    tokenize={false}
  >
    Graphic
    : 
    <Token
      className="item"
      key="0"
      tokenize={false}
    >
      <AttributionItem
        credit={Array []}
        key="0"
        source={
          Array [
            TextNode {
              "styles": Array [],
              "type": "Text",
              "value": "source 2",
            },
          ]
        }
        tokenize={false}
      />
    </Token>
  </Token>
</Fragment>
`;

exports[`<ImageAttributionComponent /> should render licensors 1`] = `
Array [
  <WithKinjaMeta
    key="0"
    nodes={
      Array [
        TextNode {
          "styles": Array [],
          "type": "Text",
          "value": "source",
        },
      ]
    }
  />,
  ", ",
  <WithKinjaMeta
    key="1"
    nodes={
      Array [
        TextNode {
          "styles": Array [],
          "type": "Text",
          "value": "source 1",
        },
      ]
    }
  />,
]
`;
